com/bl/ttt/tictactoe.py

```diff
@@ -0,0 +1,62 @@
+import re
+
+
+class Tictactoe:
+    """ Cell coordinates are:
+    0	1	2
+    3	4	5
+    6	7	8
+    """
+    def __init__(self):
+        self.cells = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+
+    def is_valid_move(self, player_id, move):
+        try:
+            if player_id == 'X' or player_id == 'O':
+                coordinate = int(move)
+                if 8 >= coordinate >= 0:
+                    if self.cells[coordinate] == ' ':
+                        valid_move_cell = coordinate
+                        return True
+
+            return False
+
+        except (ValueError, TypeError):
+            return False
+
+    def make_valid_move(self, move):
```

Comment:
I think this should be 
`def make_valid_move(self, move):
    self.cells[move.cell] = move.player_id`

Reply:
Thinking about that... and will do some better thinking after lunch and a siesta cycle :).

Reply:
I reckon this might need player_id as a second argument:
make_valid_move(self, player_id, cell_chosen):  self.cells[cell_chosen] = player_id
... because I want to pass player_id in to use it, right?

Reply:
No that is the whole point of having a TictaktoeMove class. it holds the player_id and cell_chosen for the move so they always travel together and can be passed as one object in to a function. The fewer the number of params on a function the easier is is to understand what the function does.
I think we should go through TivtactoeMove and Tictaktoe class again I must be failing to help you understand how they are related and what they are used for.

Reply:
You are right we do ned player_id to do a vailidation of the move.
Pass an instance of TictactoeMove class to the function so the player_id and the cell_chose are bound to each other when the move is made and travel together to the Tictactoe game class.

Reply:
That makes sense of course.
I will get my head and the code around that first thing in the AM.
Thanks for your patience.

Reply:
That is done. I wonder why an argument named move which I can send a class to proved so difficult to get my head around?